pkg/util/test/util.go

```diff
@@ -643,6 +655,102 @@ func validateImageTag(tag, namespace string, listOptions map[string]string) erro
 	return nil
 }
 
+func validatePodArgs(cluster *corev1alpha1.StorageCluster, pod v1.Pod) error {
+	args := pod.Spec.Containers[0].Args
+	for i := 0; i < len(args); i++ {
+		switch args[i] {
+		case "-d":
+			if cluster.Spec.Network != nil && cluster.Spec.Network.DataInterface != nil {
+				if *cluster.Spec.Network.DataInterface != args[i+1] {
+					return fmt.Errorf("data network interface does not match. expected: %s, got: %s",
+						*cluster.Spec.Network.DataInterface, args[i+1])
+				}
+			} else {
+				return fmt.Errorf("data network interface expected to be set")
+			}
+			i++
+		case "-s":
+			devices := strings.Join(*cluster.Spec.Storage.Devices, ",")
+			if !strings.Contains(devices, args[i+1]) {
+				return fmt.Errorf("expected device %s to be in %v", args[i+1], devices)
+			}
+			i++
+		case "-T":
+			// TODO validate backend storage type
+			i++
+		case "-cache":
+			// TODO validate cache
+			i++
+		case "-dedicated_cache":
+			// TODO validate dedicated cache
+		case "-j":
+			if *cluster.Spec.Storage.JournalDevice != args[i+1] {
+				return fmt.Errorf("expected journal device %s to be %v", args[i+1],
+					*cluster.Spec.Storage.JournalDevice)
+			}
+		case "-metadata":
+			if *cluster.Spec.Storage.SystemMdDevice != args[i+1] {
+				return fmt.Errorf("expected metadata device %s to be %v", args[i+1],
+					*cluster.Spec.Storage.SystemMdDevice)
+			}
+		case "-z":
+			// TODO validate storageless node
+		case "-r":
+			port, _ := strconv.ParseUint(args[i+1], 10, 32)
+			if *cluster.Spec.StartPort != uint32(port) {
+				return fmt.Errorf("expected port %d to be %d", port, *cluster.Spec.StartPort)
+			}
+		case "-m":
+			if cluster.Spec.Network != nil && cluster.Spec.Network.MgmtInterface != nil {
+				if *cluster.Spec.Network.MgmtInterface != args[i+1] {
+					return fmt.Errorf("mgmt network interface does not match. expected: %s, got: %s",
+						*cluster.Spec.Network.MgmtInterface, args[i+1])
+				}
+			} else {
+				return fmt.Errorf("data network interface expected to be set")
+			}
+			i++
+		case "-k":
+			expectedEndpoints := strings.Split(args[i+1], ",")
+			sort.Strings(expectedEndpoints)
+			existingEndpoints := cluster.Spec.Kvdb.Endpoints
```

Comment:
Args `-cert`, `-key`, `-ca`, `-acltoken` and `-userpwd` should have the values from `spec.kvdb.authSecret`.
This is different from other params because here the authSecret is just the name of the secret in cluster.Namespace. We will have to read the secret and validate the above values correspond to a key from the k8s secret.
```
key in k8s secret (arg)
kvdb-ca.crt (-ca)
kvdb.crt (-cert)
kvdb.key (-key)
acl-token (-acltoken)
username
password (-userpwd username:password)
```